2996_reflection/reflection.md

```diff
@@ -1823,6 +1819,48 @@ Change the grammar for `$operator-or-punctuator$` in paragraph 1 of [lex.operato
 ```
 :::
 
+### [basic.types.general]
+
+Change the first sentence in paragraph 9 of [basic.types.general]{.sref} as follows:
+
+::: bq
+[9]{.pnum} Arithmetic types (6.8.2), enumeration types, pointer types, pointer-to-member types (6.8.4),[ `std::meta::info`,]{.addu} `std::nullptr_t`, and cv-qualified (6.8.5) versions of these types are collectively called scalar types.
+:::
+
+Add a new paragraph at the end of [basic.types.general]{.sref} as follows:
+
+::: bq
+::: addu
+
+[*]{.pnum} A type is *consteval-only* if it is:
+
+  - `std::meta::info`
+  - an array type with a consteval-only element type
+  - a class type with a consteval-only base class type or consteval-only nonstatic data member type
+  - a pointer or reference to a consteval-only type
+  - a function type with a consteval-only return type or a consteval-only parameter type
+
+An object of consteval-only type shall either end its lifetime during the evaluation of a manifestly constant-evaluated expression or conversion ([expr.const]{.sref}), or be a constexpr variable that is not odr-used ([basic.def.odr]{.sref}).
```

Comment:
Not pressing now, but should this whole section live in [expr.const] instead?

Reply:
It does straddle [basic.types] and [expr.const], but I think I prefer it in [basic.types] because it is a type classification.